libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java

```diff
@@ -52,29 +56,21 @@
   @Override
   public boolean isUserOffRoute(Location location, RouteProgress routeProgress, MapboxNavigationOptions options) {
 
+    if (checkDistanceRemaining(routeProgress)) {
+      return true;
+    }
+
     if (!validOffRoute(location, options)) {
       return false;
     }
-
     Point currentPoint = Point.fromLngLat(location.getLongitude(), location.getLatitude());
+    boolean isOffRoute = checkOffRouteRadius(location, routeProgress, options, currentPoint);
 
-    // Get distance from the current step to future point
-    LegStep currentStep = routeProgress.currentLegProgress().currentStep();
-    double distanceFromCurrentStep = userTrueDistanceFromStep(currentPoint, currentStep);
-
-    // Create off-route radius from the max of our dynamic or accuracy based tolerances
-    double offRouteRadius = createOffRouteRadius(location, routeProgress, options, currentPoint);
-
-    // Off route if this distance is greater than our offRouteRadius
-    boolean isOffRoute = distanceFromCurrentStep > offRouteRadius;
-
-    // If not offRoute at this point, do not continue with remaining logic
     if (!isOffRoute) {
       // Even though the current point is not considered off-route, check to see if the user is
```

Comment:
Could we remove somehow these comments from here and include them into the Javadoc ☝️?